fvm/fvm.go

```diff
@@ -242,3 +233,70 @@ func (vm *VirtualMachine) GetAccount(
 	}
 	return account, nil
 }
+
+// GetAccountBalance returns an account balance by address or an error if none exists.
+func GetAccountBalance(
+	ctx Context,
+	address flow.Address,
+	storageSnapshot snapshot.StorageSnapshot,
+) (
+	uint64,
+	error,
+) {
+	storageTxn := initializeStorageTransaction(ctx, storageSnapshot)
+
+	env := environment.NewScriptEnv(
+		context.Background(),
+		ctx.TracerSpan,
+		ctx.EnvironmentParams,
+		storageTxn)
+
+	accountBalance, err := env.GetAccountBalance(common.MustBytesToAddress(address.Bytes()))
+
+	if err != nil {
+		return 0, fmt.Errorf("cannot get account balance: %w", err)
+	}
+	return accountBalance, nil
+}
+
+// GetAccountKeys returns an account keys by address or an error if none exists.
+func GetAccountKeys(
+	ctx Context,
+	address flow.Address,
+	storageSnapshot snapshot.StorageSnapshot,
+) (
+	[]flow.AccountPublicKey,
+	error,
+) {
+	storageTxn := initializeStorageTransaction(ctx, storageSnapshot)
+
+	env := environment.NewScriptEnv(
+		context.Background(),
+		ctx.TracerSpan,
+		ctx.EnvironmentParams,
+		storageTxn)
```

Comment:
This part can be in the common function as well. The `initializeStorageTransaction` could then be renamed to something like `getScriptEnvironment`

Reply:
Sorry, I missed that.

The return type should be `environment.Environment`. The new functions can be added to `environment.Environment`.